### DISCUSSION TITLE
Typescript, yes or no?

### DISCUSSION THREAD

**Discussion by timlrx** [timlrx/tailwind-nextjs-starter-blog#90] [General]
Had a few requests for typescript but I also want to get a sense of the broader user base / potential users. Should we keep it as js or  migrate to ts? Since this is a template which people clone and modify, I tried to keep it as accessible as possible i.e. js. We can consider improving the developer experience with JSDoc comments but I am not sure about fully adopting typescript.

Alternatively, if you know of ways where repo templates cater to both js and ts in a convenient manner, happy to hear about them.

**Comment by rsipakov** [comment]
You have led the arguments to keep it as JS. I vote for it.

**Comment by timlrx** [comment]
@DoctorDerek @aloisdg how about javascript files with jdoc types like [xdm](https://github.com/wooorm/xdm)? Got the inspiration from there while experimenting with it. Or are you looking for a strict typescript version of the code?

**Reply by aloisdg** [reply]
More strictness is the way to go as far as I am concerned

**Comment by DoctorDerek** [comment]
Personally I prefer TypeScript because despite its quirks there's going to
be strict type checking integrated into the build step.

It's not error proof, since undefined values can come from the Markdown
files resulting in run-time errors that TypeScript would normally prevent
(e.g. .map() is not a function), but I find it incredibly useful as a type
of self-documenting code.

Some people hate TypeScript vehemently, though. 🙂 In my experience
converting a fork of this project, everything is straightforward except the
lib/utils files, some of which would need to be rewritten.

**Reply by timlrx** [reply]
Got it, thanks. In that case, it's better to maintain a separate branch. Or alternatively, we can provide a `create-react-app` like cli for the user to choose his preferred template. That would also make the site easier to setup, though it requires quite a bit more development work.

**Reply by aloisdg** [reply]
the cli solution is what svelte folk did too

**Comment by fullStackDataSolutions** [comment]
I prefer TS, but I'd be happy with a conversion guide.
